packages/vats/src/localchain.js

```diff
@@ -0,0 +1,89 @@
+import { E } from '@endo/far';
+import { M } from '@endo/patterns';
+
+const { Fail } = assert;
+
+export const LocalChainAccountI = M.interface('LocalChainAccount', {
+  getAddress: M.callWhen().returns(M.string()),
+  executeTx: M.callWhen(M.arrayOf(M.any())).returns(M.arrayOf(M.any())),
+  // A sample of adding specific queries.
+  allBalances: M.callWhen().returns(M.any()),
+});
+
+/** @param {import('@agoric/base-zone').Zone} zone */
+const prepareLocalChainAccount = zone =>
+  zone.exoClass(
+    'LocalChainAccount',
+    LocalChainAccountI,
+    (system, address, chain) => ({ system, address, chain }),
+    {
+      // Information that the account creator needs.
+      async getAddress() {
+        return this.state.address;
+      },
+      async allBalances() {
+        // We make a balance request, scoped to our own address.
+        const { address, chain } = this.state;
+        const res = await chain.query([
+          {
+            '@type': '/cosmos.bank.v1beta1.QueryAllBalancesRequest',
+            address,
+          },
+        ]);
+        if (res[0].error) {
+          throw Fail`query failed: ${res[0].error}`;
+        }
+        return res[0].reply;
+      },
+      async executeTx(messages) {
+        const { system, address } = this.state;
+        const obj = {
```

Comment:
could be helpful to point out in a comment that this relies on the `VLOCALCHAIN_EXECUTE_TX` handler to ensure that the `messages` authenticate with the `address`.

also we should have some tests of that. I don't see them. cc @raphdev

Reply:
I tested this manually, but you're totally right; we rely on this property for security so I've added a test in vlocalchain_test.go.